beacon-chain/state/state-native/getters_withdrawal_test.go

```diff
@@ -1,597 +1,353 @@
 	t.Run("ok for deneb", func(t *testing.T) {
-		s := BeaconState{version: version.Deneb, nextWithdrawalIndex: 123}
+		s, err := state_native.InitializeFromProtoDeneb(&ethpb.BeaconStateDeneb{NextWithdrawalIndex: 123})
+		require.NoError(t, err)
 		i, err := s.NextWithdrawalIndex()
 		require.NoError(t, err)
 		assert.Equal(t, uint64(123), i)
 	})
 	t.Run("ok", func(t *testing.T) {
-		s := BeaconState{version: version.Capella, nextWithdrawalIndex: 123}
+		s, err := state_native.InitializeFromProtoCapella(&ethpb.BeaconStateCapella{NextWithdrawalIndex: 123})
+		require.NoError(t, err)
 		i, err := s.NextWithdrawalIndex()
 		require.NoError(t, err)
 		assert.Equal(t, uint64(123), i)
 	})
 	t.Run("version before Capella not supported", func(t *testing.T) {
-		s := BeaconState{version: version.Bellatrix}
-		_, err := s.NextWithdrawalIndex()
+		s, err := state_native.InitializeFromProtoBellatrix(&ethpb.BeaconStateBellatrix{})
+		require.NoError(t, err)
+		_, err = s.NextWithdrawalIndex()
 		assert.ErrorContains(t, "NextWithdrawalIndex is not supported", err)
 	})
 }
 
 func TestNextWithdrawalValidatorIndex(t *testing.T) {
 	t.Run("ok for deneb", func(t *testing.T) {
-		s := BeaconState{version: version.Deneb, nextWithdrawalValidatorIndex: 123}
+		pb := &ethpb.BeaconStateDeneb{NextWithdrawalValidatorIndex: 123}
+		s, err := state_native.InitializeFromProtoDeneb(pb)
+		require.NoError(t, err)
 		i, err := s.NextWithdrawalValidatorIndex()
 		require.NoError(t, err)
 		assert.Equal(t, primitives.ValidatorIndex(123), i)
 	})
 	t.Run("ok", func(t *testing.T) {
-		s := BeaconState{version: version.Capella, nextWithdrawalValidatorIndex: 123}
+		pb := &ethpb.BeaconStateCapella{NextWithdrawalValidatorIndex: 123}
+		s, err := state_native.InitializeFromProtoCapella(pb)
+		require.NoError(t, err)
 		i, err := s.NextWithdrawalValidatorIndex()
 		require.NoError(t, err)
 		assert.Equal(t, primitives.ValidatorIndex(123), i)
 	})
 	t.Run("version before Capella not supported", func(t *testing.T) {
-		s := BeaconState{version: version.Bellatrix}
-		_, err := s.NextWithdrawalValidatorIndex()
+		s, err := state_native.InitializeFromProtoBellatrix(&ethpb.BeaconStateBellatrix{})
+		require.NoError(t, err)
+		_, err = s.NextWithdrawalValidatorIndex()
 		assert.ErrorContains(t, "NextWithdrawalValidatorIndex is not supported", err)
 	})
 }
 
-func TestHasETH1WithdrawalCredentials(t *testing.T) {
-	creds := []byte{0xFA, 0xCC}
-	v := &ethpb.Validator{WithdrawalCredentials: creds}
-	require.Equal(t, false, hasETH1WithdrawalCredential(v))
-	creds = []byte{params.BeaconConfig().ETH1AddressWithdrawalPrefixByte, 0xCC}
-	v = &ethpb.Validator{WithdrawalCredentials: creds}
-	require.Equal(t, true, hasETH1WithdrawalCredential(v))
-	// No Withdrawal cred
-	v = &ethpb.Validator{}
-	require.Equal(t, false, hasETH1WithdrawalCredential(v))
-}
+func TestExpectedWithdrawals(t *testing.T) {
+	for _, stateVersion := range []int{version.Capella, version.Deneb, version.Electra} {
```

Comment:
I refactored TestExpectedWIthdrawals_$FORK to be a more table driven style instead of a copy and paste with the version changed.